srml/system/src/lib.rs

```diff
@@ -361,7 +361,12 @@ pub fn ensure_root<OuterOrigin, AccountId>(o: OuterOrigin) -> Result<(), &'stati
 	}
 }
 
-/// Ensure that the origin `o` represents an unsigned extrinsic. Returns `Ok` or an `Err` otherwise.
+/// Ensure that the origin `o` represents an unsigned extrinsic.
+///
+/// Returns `Ok` or an `Err` otherwise.
+/// Use this method to verify extrinsics added by block producers.
+/// If you runtime uses unsigned transactions that are propagated in the network
+/// rather use `ensure_unsigned`.
```

Comment:
Ideally, passing an unsigned transaction to `ensure_inherent` would result in a compile-time error.  Can this be enforced?

Reply:
Not in the slightest.